### PR TITLE
Snapshotcli

### PR DESCRIPTION
**Pull request by hrudaya21**
Replica Snapshot CLI Commands Implemented

- Create Replica Snapshot
- List Replica Snapshot
- Delete Replica Snapshot

### CODE REVIEW

File: io-engine/src/bin/io-engine-client/v1/replica_cli.rs [hunk 1]
```diff
@@ -139,6 +184,13 @@ pub async fn handler(
         ("share", Some(args)) => replica_share(ctx, args).await,
         ("unshare", Some(args)) => replica_unshare(ctx, args).await,
         ("stats", Some(args)) => replica_stat(ctx, args).await,
+        ("snapshot_create", Some(args)) => {
+            replica_snapshot_create(ctx, args).await
+        }
+        ("snapshot_list", Some(args)) => replica_snapshot_list(ctx, args).await,
```
Comment (mtzaurus) [inline io-engine/src/bin/io-engine-client/v1/replica_cli.rs:190]: Snapshot removal seems to be a generic snapshot operation, so can we introduce a dedicated CLI command for it ? We could later generalise snapshot listing too.
```
> io-engine-client snapshot delete <UUID>
```

We could later also generalise our .proto files and move such generic operations there (currently snapshot listing/removal reside in replica-related .proto). So let's have CLI properly formed first.

Comment (hrudaya21) [reply io-engine/src/bin/io-engine-client/v1/replica_cli.rs:190]: Yes.. I have created a task already to handle the isolation of common operations. For the first task, We can put all CLI works. Following is the task to track.
https://dcsw-bolt.atlassian.net/browse/GTM-862

**Review by mtzaurus — APPROVED** [review]

**Review by dsharma-dc — APPROVED** [review]

**Comment by hrudaya21** [comment]
bors try

**Comment by hrudaya21** [comment]
bors merge

**Pull request merged**